packages/cozy-client/src/types.js

```diff
@@ -380,4 +380,26 @@ import { QueryDefinition } from './queries/dsl'
  * @property {string} [codeChallenge]
  */
 
+/**
+ * Subdomain type for a Cozy. Can be flat or nested subdomains
+ *
+ * Example of 'flat' domain: https://claude-notes.somedomain.fr
+ * Example of 'nested' domain: https://notes.claude.somedomain.fr
+ *
+ * @typedef {'flat'|'nested'} SubdomainType
+ */
+
+/**
+ * Represents the differents parts of a deconstructed Cozy link
+ *
+ * @typedef {object} CozyLinkData
```

Comment:
nit: Why not calling this CozyLink or CozyLinkParams or FlattenCozyUrl?

Reply:
I'm wondering if `CozyLink` would bring confusion to what it contains. For me a link is a `string` or a `URL` object, I would not expect it to contain structured data. That said, the `URL` object contains those kind of data, so I'm not sure anymore.

Regarding `Params`, those are not params, but the description of the entire link. So I would not use this term.

`SplittedCozyUrl` would work maybe, but here again I'm wondering if someone can expect to find an array here like for the output from `string.split()`.

Maybe we should use `CozyLink` as you suggested but using a full class that allows to manipulate a link like for `URL`?

Reply:
After the whole discussion, I would go for CozyLinkData